lib/MonadicRewrite.thy

```diff
@@ -32,6 +32,10 @@ lemma monadic_rewrite_refl_generic:
 
 lemmas monadic_rewrite_refl = monadic_rewrite_refl_generic[where P=\<top>]
 
+lemma monadic_rewrite_sym:
+  "monadic_rewrite False True P f g \<Longrightarrow> monadic_rewrite False True P g f"
+  by (simp add: monadic_rewrite_def)
```

Comment:
This is really high up, given the `F E` flags are explicitly set, so it's a sym rule only for that case. Does it still hold if the `F` flag is left generic?

Reply:
I'm happy to move this. Where would be best, do you think?

Unfortunately, I couldn't see any generalisation of this for either the `F` or `E` flag

Reply:
Well, if this is the only way symmetry works, then leave it where it is. If we have doubts, or there's something we're paving over, maybe dump it at the end of this section after `monadic_rewrite_weaken2`?

Reply:
I've taken another look and I'm sure that we can't generalise this any further, so I think I might leave it where it is